components/script/script_thread.rs

```diff
@@ -2018,40 +1967,9 @@ impl ScriptThread {
             docs.clear();
         }
 
-        // https://html.spec.whatwg.org/multipage/#event-loop-processing-model step 7.12
-
-        // Issue batched reflows on any pages that require it (e.g. if images loaded)
-        // TODO(gw): In the future we could probably batch other types of reflows
-        // into this loop too, but for now it's only images.
-        debug!("Issuing batched reflows.");
-        for (_, document) in self.documents.borrow().iter() {
-            // Step 13
-            if !document.is_fully_active() {
-                continue;
-            }
-            let window = document.window();
-
-            let _realm = enter_realm(&*document);
-
-            window
-                .upcast::<GlobalScope>()
-                .perform_a_dom_garbage_collection_checkpoint();
-
-            let pending_reflows = window.get_pending_reflow_count();
-            if pending_reflows > 0 {
-                window.reflow(ReflowGoal::Full, ReflowReason::PendingReflow, can_gc);
-            } else {
-                // Reflow currently happens when explicitly invoked by code that
-                // knows the document could have been modified. This should really
-                // be driven by the compositor on an as-needed basis instead, to
-                // minimize unnecessary work.
-                window.reflow(
-                    ReflowGoal::Full,
-                    ReflowReason::MissingExplicitReflow,
-                    can_gc,
-                );
-            }
-        }
+        // If any message triggered a rendering opportunity, then update the rendering now.
+        self.update_the_rendering(compositor_requested_update_the_rendering, can_gc);
+        self.perform_a_microtask_checkpoint(can_gc);
```

Comment:
Note sure if it could be noticeable, but we should make this conditional on the `update_the_rendering` steps actually having run because this should only happen after a task(which running the steps are simulating here). Can be a TODO with a link to https://github.com/servo/servo/issues/32003 for now.

Reply:
Okay. I've gone ahead and done that here.